packages/data-service/src/graphqlObservable.ts

```diff
@@ -0,0 +1,317 @@
+    }, Observable.of([]));
+  }
+
+  function getField(
+    parentType: GraphQLType | null,
+    definition: SchemaNode
+  ): GraphQLField<any, any> | null {
+    // Go one level deeper into the query
+    if (parentType instanceof GraphQLObjectType && isFieldNode(definition)) {
+      const parentFields = parentType.getFields();
+      const fieldName = definition.name.value;
+
+      if (parentFields[fieldName]) {
+        return parentFields[definition.name.value];
+      }
+    }
+
+    // These cases should ideally at some point be not existant,
+    // but due to our partial implementation this loop-hole is needed
+    return null;
+  }
+
+  function getResultType(
+    parentType: GraphQLType | null,
+    definition: SchemaNode,
+    instance: any
+  ): GraphQLType | null {
+    const translateOperation = {
+      query: "Query",
+      mutation: "Mutation"
+    };
+
+    // Operation is given (query or mutation), returns a type
+    if (isOperationDefinition(definition)) {
+      return types[translateOperation[definition.operation]];
+    }
+
+    // Get one level deeper in the query nesting
+    const field = getField(parentType, definition);
+    if (field !== null) {
+      const fieldType = getNamedType(field.type);
+
+      // Make this abstract type concrete if possible
+      if (
+        fieldType instanceof GraphQLInterfaceType &&
+        fieldType.resolveType instanceof Function
+      ) {
+        // We currenlty only allow resolveType to return a GraphQLObjectType
+        // and we pass in the wrong values as we don't need this feature currently
+        // @ts-ignore
+        return getNamedType(fieldType.resolveType(instance));
+      } else {
+        return fieldType;
+      }
+    }
+
+    return null;
+  }
+
+  return resolve(doc.definitions[0], context, null, null).map((data: T) => ({
+    data
+  }));
+}
+
+function throwObservable(error: string): Observable<any> {
+  const graphqlErrorMessage = `graphqlObservable error: ${error}`;
+  const graphqlError = new Error(graphqlErrorMessage);
+
+  return Observable.throw(graphqlError);
+}
+
+function buildResolveArgs(definition: FieldNode, context: object) {
+  return (definition.arguments || []).reduce(
+    (carry, arg) => ({
+      ...carry,
+      ...(arg.value.kind === Kind.VARIABLE
+        ? // @ts-ignore
+          { [arg.name.value]: context[arg.value.name.value] }
+        : { [arg.name.value]: arg.value.value })
+    }),
+    {}
+  );
+}
+
+const objectAppendWithKey = (key: string) => {
+  return (destination: object, source: any) => ({
+    ...destination,
+    [key]: source
+  });
+};
+
+const listAppend = <T = any>(destination: T[], source: T) => {
+  return destination.concat(source);
+};
+
+function resolveField(
+  field: GraphQLField<any, any, { [argName: string]: any }>,
+  definition: FieldNode,
+  context: object,
+  parent: any
+) {
+  if (!isFieldWithResolver(field)) {
+    return Observable.of(parent[field.name]);
```

Comment:
I wonder if it would simplify things slightly to provide this as a default field resolver `parent => Observable.of(parent[field.name])`

so instead of defining and checking `isFieldWithResolver(field)` you could define `getFieldResolver(field)` which returns `field.resolver || parent => Observable.of(parent[field.name])`

Reply:
I think we are going to have a similar problem with polymorphisms, I would like to see which version is clearer in the end. I hope this is okay for you 😇